### PR TITLE
Fixes for spacing and save

### PR DESCRIPTION
**Pull request by harshad1**
This is a follow up PR for the reduction in spacing. Due to the way padding and minWidth interacted, the extra icon spacing would not take effect until the extra icon spacing was > 8. In this PR, I have fixed it so spacing is calculated and applied correctly.

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java [hunk 1]
```diff
@@ -663,15 +663,14 @@ public String getFragmentTag() {
     // Save the file
     // Only supports java.io.File. TODO: Android Content
     public boolean saveDocument(final boolean forceSaveEmpty) {
-        if (!isAdded()) {
+        if (!isAdded() || _hlEditor == null || _document == null) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java:666]: I've never seen this fail on my phone, but I noticed it occasionally failing in the emulator, so I added some defensive code.

File: app/src/main/res/layout/quick_keyboard_button.xml [hunk 2]
```diff
@@ -11,17 +11,15 @@
...
-    android:layout_height="wrap_content"
+    android:layout_width="24dp"
+    android:layout_height="24dp"
     android:background="@drawable/keyboard_shortcut_button"
     android:gravity="center_vertical|center_horizontal"
-    android:minWidth="40dp"
-    android:minHeight="48dp"
-    android:paddingStart="12dp"
-    android:paddingLeft="12dp"
+    android:paddingLeft="8dp"
```
Comment (harshad1) [inline app/src/main/res/layout/quick_keyboard_button.xml:18]: 24dp icon + (8 * 2 = 16) dp padding = 40dp min width

File: app/src/main/res/layout/quick_keyboard_button.xml [hunk 3]
```diff
@@ -11,17 +11,15 @@
...
+    android:paddingRight="8dp"
     android:paddingTop="8dp"
-    android:paddingEnd="12dp"
-    android:paddingRight="12dp"
     android:paddingBottom="8dp"
+    android:scaleType="fitCenter"
     android:tint="@color/textActionIconColor"
+    android:src="@drawable/ic_add_white_24dp"
```
Comment (harshad1) [inline app/src/main/res/layout/quick_keyboard_button.xml:24]: Default drawable makes it easier to debug the layout. will not show up in practice.

Comment (gsantner) [reply app/src/main/res/layout/quick_keyboard_button.xml:24]: tools:src ?

Comment (harshad1) [reply app/src/main/res/layout/quick_keyboard_button.xml:24]: Anything, honestly. Just so that there is a default icon I can see in the layout :)

Comment (gsantner) [reply app/src/main/res/layout/quick_keyboard_button.xml:24]: then tools: instead android: is exactly whats needed

**Review by gsantner — APPROVED** [review]

**Pull request merged**